wolk/service/registration_service/RegistrationService.cpp

```diff
@@ -140,23 +149,108 @@ std::unique_ptr<ErrorMessage> RegistrationService::removeDevices(const std::stri
     {
         const auto errorMessage = "Failed to generate the outgoing message.";
         LOG(ERROR) << errorPrefix << " -> " << errorMessage;
-        return std::unique_ptr<ErrorMessage>{
-          new ErrorMessage{deviceKey, errorMessage, std::chrono::system_clock::now()}};
+        return false;
     }
 
     // Send the message out
     if (!m_connectivityService.publish(message))
     {
         const auto errorMessage = "Failed to send the outgoing message.";
         LOG(ERROR) << errorPrefix << " -> " << errorMessage;
-        return std::unique_ptr<ErrorMessage>{
-          new ErrorMessage{deviceKey, errorMessage, std::chrono::system_clock::now()}};
+        return false;
+    }
+    return true;
+}
+
+std::shared_ptr<std::vector<std::string>> RegistrationService::obtainChildren(const std::string& deviceKey,
+                                                                              std::chrono::milliseconds timeout)
+{
+    LOG(TRACE) << METHOD_INFO;
+    const auto errorPrefix = "Failed to obtain children";
+
+    // Parse the message
+    auto message =
+      std::shared_ptr<Message>{m_protocol.makeOutboundMessage(deviceKey, ChildrenSynchronizationRequestMessage{})};
+    if (message == nullptr)
+    {
+        LOG(ERROR) << errorPrefix << " -> Failed to generate outgoing `ChildrenSynchronizationRequestMessage`.";
+        return nullptr;
+    }
+
+    // Lock the mutex for the map, just so the response arrival can't be faster than us putting the query data into the
+    // map.
+    std::atomic_bool called{false};
+    auto list = std::make_shared<std::vector<std::string>>();
+    {
+        std::lock_guard<std::mutex> lock{m_childrenSyncDevicesMutex};
+        // Send out the message
+        if (!m_connectivityService.publish(message))
+        {
+            LOG(ERROR) << errorPrefix << " -> Failed to send the outgoing `ChildrenSynchronizationRequestMessage`.";
+            return nullptr;
+        }
+        if (m_queries.find(deviceKey) == m_queries.cend())
+            m_queries.emplace(deviceKey, std::queue<std::function<void(std::vector<std::string>)>>{});
+        auto weakList = std::weak_ptr<std::vector<std::string>>{list};
+        m_queries[deviceKey].push([this, &called, weakList](const std::vector<std::string>& children) {
```

Comment:
This m_queries is never accesed and callback is never called. And if it was called after this method returns it would access the stack allocated atomic which would be destroyed